lib/logster/base_store.rb

```diff
@@ -84,6 +84,21 @@ def check_rate_limits(severity)
       not_implemented
     end
 
+    # takes a string as `pattern` and places it under the set `set_name`
+    def insert_pattern(set_name, pattern)
+      not_implemented
+    end
+
+    # takes a string as `pattern` and removes it from the set `set_name`
+    def remove_pattern(set_name, pattern)
+      not_implemented
+    end
+
+    # returns an array of strings each of which must be convertible to regexp
+    def get_patterns(set_name)
+      not_implemented
```

Comment:
I don't mind this pattern, some people complain raising https://ruby-doc.org/core-2.2.0/NotImplementedError.html is a misuse which is true, so this is a good enough clean workaround imo